spec/controllers/users/phone_confirmation_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+require 'rails_helper'
+
+describe Users::PhoneConfirmationController, devise: true do
+  describe 'before_actions' do
+    it 'includes authentication checks' do
+      expect(subject).to have_filters(
+        :before,
+        :authenticate_user!,
+        :check_for_unconfirmed_mobile
+      )
+    end
+  end
+
+  describe '#send_code' do
+    before do
+      sign_in_as_user
+    end
+
+    context 'when :unconfirmed_mobile is not set in session' do
+      it 'redirects to rool_url' do
+        get :send_code
+
+        expect(response).to redirect_to(root_url)
+      end
+    end
+
+    context 'when :unconfirmed_mobile is set in session' do
+      before do
+        subject.user_session[:unconfirmed_mobile] = '+1 (555) 555-5555'
+      end
+
+      it 'generates a confirmation code in the session' do
+        expect(subject.user_session[:phone_confirmation_code]).to be_nil
+        # We're testing an implementation detail here, but its an important one.
+        expect(SecureRandom).to receive(:random_number).with(10**Devise.direct_otp_length)
+        get :send_code
```

Comment:
Please add empty lines before and after the `get` line to separate the expectations from the rest of the test.